### PR TITLE
[Tests] Tidy up REST interface functional tests

### PR DESCRIPTION
**Pull request by romanz**
Following the comment at https://github.com/bitcoin/bitcoin/pull/12717#pullrequestreview-106189117.

### CODE REVIEW

**Comment by romanz** [comment]
@jnewbery Please let me know what else could be improved in the tests :)

**Comment by jnewbery** [comment]
Will review after #12717 is merged (feel free to ping me again once that happens!)

**Comment by laanwj** [comment]
> Will review after #12717 is merged (feel free to ping me again once that happens!)

It's merged, please rebase (so that this is only test changes).

File: test/functional/interface_rest.py [hunk 1]
```diff
@@ -4,351 +4,297 @@
...
 
-        #check if the 3 tx show up in the new block
-        json_string = http_get_call(url.hostname, url.port, '/rest/block/'+newblockhash[0]+self.FORMAT_SEPARATOR+'json')
-        json_obj = json.loads(json_string)
-        for tx in json_obj['tx']:
-            if not 'coinbase' in tx['vin'][0]: #exclude coinbase
-                assert_equal(tx['txid'] in txs, True)
+        # Check if the 3 tx show up in the new block
+        json_obj = self.test_rest_request("/block/{}".format(newblockhash[0]))
+        non_coinbase_txs = [tx['txid'] for tx in json_obj['tx']
```
Comment (jnewbery) [inline test/functional/interface_rest.py]: I'm not sure if ordering is guaranteed here. Perhaps construct sets and check for set equality?

**Review by jnewbery — COMMENTED** [review]
Tested ACK d54e3e704838eb39a5e6e33d8b530c6ed404fd49. One comment inline.

Commits can probably be squashed a bit before merge. Feel free to take my changes without the commit messages if that makes things more structured.

File: test/functional/interface_rest.py (hunk 1 above)
Comment (romanz) [reply test/functional/interface_rest.py]: Sounds good - done.

**Comment by romanz** [comment]
> Commits can probably be squashed a bit before merge. 

Squashed most of the small commits into a few large ones.

**Comment by jnewbery** [comment]
Tested ACK 55efc1f62cef7e19efcb58ba68fbc2053bfa19a0.

Nicely structured PR. Very easy to review! Thanks

**Pull request merged**